swift/Samples/Tutorial/Tutorial1.md

```diff
@@ -0,0 +1,288 @@
+import WorkflowUI
+
+
+public final class TutorialContainerViewController: UIViewController {
+    let containerViewController: UIViewController
+
+    public init() {
+        // Create a view registry. This will allow the infrastructure to map `Screen` types to their respective view controller type.
+        var viewRegistry = ViewRegistry()
+        // Register the `WelcomeScreen` and view controller with the convenience method the template provided.
+        viewRegistry.registerWelcomeScreen()
+
+        // Create a `ContainerViewController` with the `WelcomeWorkflow` as the root workflow, with the view registry we just created.
+        containerViewController = ContainerViewController(
+            workflow: WelcomeWorkflow(),
+            viewRegistry: viewRegistry)
+
+        super.init(nibName: nil, bundle: nil)
+    }
+```
+
+We did a few things here.
+- First, we created a *view registry* that is used to map screen types to view controllers. The view registry is used by the container to determine what view controller to instantiate the first time we see a screen.
+- Then, we registered our `WelcomeScreen` with the view registry so we have that mapping.
+- Finally, we create our *ContainerViewController* with the WelcomeWorkflow as the root.
+
+We can finally run the app again! It will look the exact same as before, but now powered by our workflow.
+
+## Driving the UI from Workflow State
+
+Right now, the workflow isn't handling any of the events from the UI. Let's update it to be responsible for the login name as well as the action when the login button is pressed.
+
+### State
+
+All workflows have a `State` type that represents the internal state of the workflow. This should be all of the data that this workflow is _responsible_ for - usually corresponds to the state for the UI.
+
+We will model the first part of state that we want to track, the login `name`. Update the `State` type to include a name. We will also need to update `makeInitialState` to give an initial value:
+```swift
+// MARK: State and Initialization
+
+extension WelcomeWorkflow {
+
+    struct State {
+        var name: String
+    }
+
+    func makeInitialState() -> WelcomeWorkflow.State {
+        return State(name: "")
+    }
+
+    // ...
+```
+
+Now that we have the state modelled, we'll send it to the UI every time a render pass happens - the text field will overwrite it's value with whatever was provided.
+
+```swift
+// MARK: Rendering
+
+extension WelcomeWorkflow {
+
+    typealias Rendering = WelcomeScreen
+
+    func render(state: WelcomeWorkflow.State, context: RenderContext<WelcomeWorkflow>) -> Rendering {
+        return WelcomeScreen(
+            name: state.name,
+            onNameChanged: { _ in
+            },
+            onLoginTapped: {
+            })
+    }
+}
+```
+
+If you run the app again, you'll see that it still behaves the same letting your type into the name field. This is because we only have rendered a screen once.
```

Comment:
Typo, missing comma.

> If you run the app again, you'll see that it still behaves the same, letting you type into the name field. This is because we have only rendered a screen once.